test/apps/iodemo/ucx_wrapper.h

```diff
@@ -27,7 +27,15 @@
 /* Forward declarations */
 class UcxConnection;
 struct ucx_request;
-struct UcxAmDesc;
+// Holds details of arrived AM message
+struct UcxAmDesc {
+    UcxAmDesc(void *data, const ucp_am_recv_param_t *param) :
+        _data(data), _param(param) {
+    }
+
+    void                         *_data;
+    const ucp_am_recv_param_t    *_param;
+};
```

Comment:
these details were supposed to be hidden in UCX wrapper

Reply:
The reason why expose UcxAmDesc to io_demo.cc is because we need to decide which type of buffer should be used based on _param->recv_attr and assign _data to buffer pointer if _param->recv_attr has UCP_AM_RECV_ATTR_FLAG_RNDV bit clear in dispatch_am_message() and handle_io_am_write_request().

Reply:
can we do that using some auxiliary function of UCX wrapper?

Reply:
Sure. We can add some functions to access the struct instead of manipulating the struct directly.

Reply:
@brminich please help check the latest commit. Thanks.